packages/core/test/client.test.js

```diff
@@ -1,5 +1,4 @@
-const { describe, it, expect, fail, spyOn } = global
```

Comment:
Converting this file to TS is a little more tricky.  When using a relative import of a JS file such as `'../client'`, TS looks for type declarations in `../client.d.ts`. Since it doesn't exist it infers it's own from the JS file itself. This is largely insufficient and these tests would have a lot of forced casting.

On the flip side we also want to test out the manually maintained definitions. This can be achieved by importing from the root with `import { Client } from '..'`. The problem is that the public definitions do not contain all the internal types that of often manipulated in this tests e.g. (`client._sessionDelegate = sessionDelegate`) so again some casting would be needed.

Needs some more though on how we're going to handle this one.

Reply:
I would be in favour of importing the public types (so they get fully tested) and then writing ad-hoc types for the private internals in the test file. Does that sound ok to you?